src/calendar/day.tsx

```diff
@@ -0,0 +1,90 @@
+import cx from 'classnames'
+import formatDate from 'date-fns/format'
+import React, { FC, ReactElement, SyntheticEvent } from 'react'
+
+type IMonthDayRenderProp = (props: IMonthDayRenderPropArgs) => ReactElement
+
+interface IMonthDayRenderPropArgs extends IProps {
+  // TODO: remove this and leave it to the passed function to format the date in the desired format
+  children: string
+}
+
+interface IProps {
+  blockClassName: string
+  customRender: IMonthDayRenderProp
+  date: string
+  handleOnClick: (event: SyntheticEvent<HTMLElement>) => void
+  handleOnEnter: (event: SyntheticEvent<HTMLElement>) => void
+  isCurrentMonth: boolean
+  isDisabled: boolean
+  isHighlighted: boolean
+  isMonthNext: boolean
+  isMonthPrev: boolean
+  isNonSelectable: boolean
+  isSelectable: boolean
+  isSelected: boolean
+  isSelectionEnd: boolean
+  isSelectionStart: boolean
+  isToday: boolean
+  isWeekend: boolean
+  isWorkday: boolean
+}
+
+const Day: FC<IProps> = (props) => {
+  const {
+    blockClassName,
+    customRender,
+    date,
+    handleOnClick,
+    handleOnEnter,
+    isCurrentMonth,
+    isDisabled,
+    isHighlighted,
+    isMonthNext,
+    isMonthPrev,
+    isNonSelectable,
+    isSelectable,
+    isSelected,
+    isSelectionEnd,
+    isSelectionStart,
+    isToday,
+    isWeekend,
+    isWorkday,
+  } = props
+  const children = formatDate(date, 'D')
+
+  if (customRender) {
+    return customRender({
+      ...props,
+      // TODO: remove this and leave it to the passed function to format the date in the desired format
+      children,
```

Comment:
@havenchyk 
I think in the future, we should only pass non formatted date to the render prop, and the render prop itself will have the full freedom to format prior to implemention

Reply:
it sounds good, initially I wanted to keep backward compatibility